cirq/google/op_deserializer.py

```diff
@@ -84,42 +87,48 @@ def __init__(self,
+        json_format.ParseDict(proto, msg)
+        return self.from_proto(msg)
+
+    def from_proto(self, proto: v2.program_pb2.Operation) -> ops.GateOperation:
+        """Turns a cirq.api.google.v2.Operation proto into a GateOperation."""
+        qubits = [devices.GridQubit.from_proto_id(q.id) for q in proto.qubits]
+        args = self._args_from_proto(proto)
         if self.num_qubits_param is not None:
             args[self.num_qubits_param] = len(qubits)
         gate = self.gate_constructor(**args)
         return gate.on(*qubits)
 
-    def _args_from_proto(self, args_proto_dict: Dict
+    def _args_from_proto(self, proto: v2.program_pb2.Operation
                         ) -> Dict[str, arg_func_langs.ArgValue]:
         return_args = {}
         for arg in self.args:
-            if arg.serialized_name not in args_proto_dict and arg.required:
+            if arg.serialized_name not in proto.args and arg.required:
                 raise ValueError(
                     'Argument {} not in deserializing args, but is required.'.
                     format(arg.serialized_name))
 
             value = None  # type: Optional[arg_func_langs.ArgValue]
-            if arg.serialized_name in args_proto_dict:
-                if 'arg_value' in args_proto_dict[arg.serialized_name]:
-                    arg_value = args_proto_dict[
-                        arg.serialized_name]['arg_value']
-                    if 'float_value' in arg_value:
-                        value = float(arg_value['float_value'])
-                    elif 'bool_values' in arg_value:
-                        value = arg_value['bool_values']['values']
-                    elif 'string_value' in arg_value:
-                        value = str(arg_value['string_value'])
-
-                if 'symbol' in args_proto_dict[arg.serialized_name]:
-                    value = sympy.Symbol(
-                        args_proto_dict[arg.serialized_name]['symbol'])
+            if arg.serialized_name in proto.args:
+                arg_proto = proto.args[arg.serialized_name]
+                which = arg_proto.WhichOneof('arg')
+                if which == 'arg_value':
+                    arg_value = arg_proto.arg_value
+                    which = arg_value.WhichOneof('arg_value')
+                    if which == 'float_value':
+                        value = float(arg_value.float_value)
+                    elif which == 'bool_values':
+                        value = arg_value.bool_values.values
+                    elif which == 'string_value':
+                        value = str(arg_value.string_value)
+                elif which == 'symbol':
+                    value = sympy.Symbol(arg_proto.symbol)
 
             if value is None and arg.required:
                 raise ValueError(
                     'Could not get arg {} from arg_proto {}'.format(
-                        arg.serialized_name, args_proto_dict))
+                        arg.serialized_name, args.proto))
```

Comment:
`args.proto` -> `proto.args`?
I think this is what mypy and pylint are crying about.

Reply:
Thanks; fixed.